lib/linter/linter.js

```diff
@@ -511,6 +488,126 @@ function getDirectiveComments(ast, ruleMapper, warnInlineConfig) {
     };
 }
 
+/**
+ * Parses comments in file to extract file-specific config of rules, globals
+ * and environments and merges them with global config; also code blocks
+ * where reporting is disabled or enabled and merges them with reporting config.
+ * @param {SourceCode} sourceCode The SourceCode object to get comments from.
+ * @param {function(string): {create: Function}} ruleMapper A map from rule IDs to defined rules
+ * @returns {{configuredRules: Object, enabledGlobals: {value:string,comment:Token}[], exportedVariables: Object, problems: LintMessage[], disableDirectives: DisableDirective[]}}
+ * A collection of the directive comments that were found, along with any problems that occurred when parsing
+ */
+function getDirectiveCommentsForFlatConfig(sourceCode, ruleMapper) {
+    const configuredRules = {};
+    const enabledGlobals = Object.create(null);
+    const exportedVariables = {};
```

Comment:
These three are unused. I think this function should now return only `{ problems, disableDirectives }`.

Reply:
Ah, right you are!